app/src/main/java/com/isc/hermes/model/navigation/UserRouteTracker.java

```diff
@@ -137,4 +154,34 @@ public boolean isUserInSegment(LatLng userLocation) {
     public JSONObject getRouteInformation() {
         return routeInformation;
     }
+
+    public double getUnvisitedRouteSize() {
+        return routeDistanceHandler.getDistance();
+    }
+
+    public boolean hasUserMoved() {
+        return userLocationTracker.hasUserMoved();
+    }
+
+    private void recoverTrack(boolean deepAttempt) {
+        if (deepAttempt) {
+            trackRecoveryProtocol.attemptDeepRecovery();
+        } else {
+            trackRecoveryProtocol.attemptSimpleRecovery(routeSegmentIndex);
+        }
+
+        if (trackRecoveryProtocol.isAttemptSuccessful()) {
+            currentSegment = trackRecoveryProtocol.getTrack();
+            routeSegmentIndex = routeSegments.indexOf(currentSegment);
+            routeDistanceHandler.update(routeSegmentIndex);
+            System.out.printf("User on track #%d\n", routeSegmentIndex);
```

Comment:
should implement log statements (Log.i for information and Log.e for error) to provide more meaningful messages. The use of TAG allows easy filtering and identification of log messages in Logcat.